crates/uv-configuration/src/dev.rs

```diff
@@ -1,445 +1,350 @@
-use std::borrow::Cow;
-
-use either::Either;
+use std::{borrow::Cow, sync::Arc};
 
 use uv_normalize::{GroupName, DEV_DEPENDENCIES};
 
-#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
-pub enum DevMode {
-    /// Include development dependencies.
-    #[default]
-    Include,
-    /// Exclude development dependencies.
-    Exclude,
-    /// Only include development dependencies, excluding all other dependencies.
-    Only,
-}
-
-impl DevMode {
-    /// Returns `true` if the specification allows for production dependencies.
-    pub fn prod(&self) -> bool {
-        matches!(self, Self::Exclude | Self::Include)
-    }
-
-    /// Returns `true` if the specification only includes development dependencies.
-    pub fn only(&self) -> bool {
-        matches!(self, Self::Only)
-    }
-
-    /// Returns the flag that was used to request development dependencies.
-    pub fn as_flag(&self) -> &'static str {
-        match self {
-            Self::Exclude => "--no-dev",
-            Self::Include => "--dev",
-            Self::Only => "--only-dev",
-        }
-    }
-
-    /// Returns `true` if the group is `dev`, and development dependencies should be included.
-    pub fn contains(&self, group: &GroupName) -> bool {
-        match self {
-            DevMode::Exclude => false,
-            DevMode::Include | DevMode::Only => group == &*DEV_DEPENDENCIES,
-        }
-    }
-}
-
-#[derive(Default, Debug, Clone)]
-pub struct DevGroupsSpecification {
-    /// Legacy option for `dependency-groups.dev` and `tool.uv.dev-dependencies`.
-    ///
-    /// Requested via the `--dev`, `--no-dev`, and `--only-dev` flags.
-    dev: Option<DevMode>,
+/// Manager of all dependency-group decisions and settings history.
+///
+/// This is an Arc mostly just to avoid size bloat on things that contain these.
+#[derive(Debug, Default, Clone)]
+pub struct DevGroupsSpecification(Arc<DevGroupsSpecificationInner>);
 
-    /// The groups to include.
+/// Manager of all dependency-group decisions and settings history.
+#[derive(Debug, Default, Clone)]
+pub struct DevGroupsSpecificationInner {
+    /// Groups to include.
+    include: IncludeGroups,
+    /// Groups to exclude (always wins over include).
+    exclude: Vec<GroupName>,
+    /// Whether an `--only` flag was passed.
     ///
-    /// Requested via the `--group` and `--only-group` options.
-    groups: Option<GroupsSpecification>,
+    /// If true, users of this API should refrain from doing "non-group things".
```

Comment:
"non-group things" not obvious to me. What's that mean?

Is this what we're chatting about on Discord? Should this just be "exclude_project" or something?

Reply:
"users of this API" is also a little confusing, idk who that is in this context

Reply:
Yeah basically, this feeds into `.prod()` which is a name that has survived *several* refactors and baffled me for a while (didn't want to churn it).

Reply:
I guess regarding this name and "prod" churn, maybe this should be like `ProjectMode` with included and excluded variants as an approximate corollary to `DevMode`? Fine to pursue separately. Here, I'm mostly looking for some clarity in the comments.

Reply:
I clarified the comments, I agree more followup renames would be good here.